ipld/routing.ipldsch

```diff
@@ -4,20 +4,39 @@ type Envelope union {
      | GetP2PProvideResponse "get-p2p-provide-response"
 } representation keyed
 
+type Multihash Bytes
+
 type GetP2PProvideRequest struct {
-     key Bytes
+     Multihashes [Multihash]
 }
 
 type GetP2PProvideResponse struct {
-     providers [Provider]
+     MultihashResults [MultihashResult]
+}
+
+type MultihashResult struct {
+     Multihash Multihash
+     ProviderResults [ProviderResult]
+}
+
+type ProviderResult struct {
+     Metadata Metadata
+     Provider Provider
+}
```

Comment:
@willscott I wanted to clarify + document some things I understood from our conversation today.

1. This schema is representing how the indexers are currently storing their data internally
2. The system is trying to be agnostic about the types of records it supports and as a result has defined the things it cares about (e.g. who published who has the data rather than who has the data) while relegating everything else to the metadata field
3. The metadata field then becomes the primary point of interest for us. Could you or @gammazero give a (even if rough) schema for what the current metadata things look like?

Reply:
Additionally, following up from discussions today the metadata schema seems like it'll be important within the indexer as well as once you separate the "who published these records" from "who has these records" you'll want some way to evaluate if the records are good for your own reputation system purposes which means understanding some (even if not all) of the record types.

Reply:
@aschmahmann 
1. The schema is representing an answer to a client asking, "who has the data".  Who published this record is not part of this, and that is a separate concern from delegated routing and responses to clients.
2. Only the _metadata_ is trying to be agnostic about what records it supports since that is consumed mostly by the provider (who has the data) to determine where/how to retrieve that data.
3. The metadata contains a "protocol" field which determines the protocol used to retrieve data (graphsync, bitswap).  This is used by the client that wants the data.  The remainder of the metadata is a payload that is used only by the provider.  It is likely something like a deal ID, but could be anything (e.g. record key for their internal database) that tells the provider how to find and retrieve the data.